db/background_mgr_resync_dcp.go

```diff
@@ -217,6 +228,41 @@ func (r *ResyncManagerDCP) Run(ctx context.Context, options map[string]interface
 	return nil
 }
 
+func getCollectionIdsFromBody(requestBody []byte, db *Database, bucket *base.GocbV2Bucket) ([]uint32, bool, error) {
+	var collectionsMap resyncPostReqBody
+	if len(requestBody) != 0 {
+		if err := json.Unmarshal(requestBody, &collectionsMap); err != nil {
+			return nil, false, fmt.Errorf("failed to unmarshal request body: %w", err)
+		}
+	}
+
+	collectionIDs := make([]uint32, 0)
+	var hasAllCollections bool
+	if len(collectionsMap) == 0 {
+		hasAllCollections = true
+		for collectionID := range db.CollectionByID {
+			collectionIDs = append(collectionIDs, collectionID)
+		}
+	} else {
+		hasAllCollections = false
+		collectionManifest, err := bucket.GetCollectionManifest()
+		if err != nil {
+			return nil, hasAllCollections, fmt.Errorf("failed to load collection manifest: %w", err)
+		}
+
+		for scopeName, collNames := range collectionsMap {
+			for _, collName := range collNames {
```

Comment:
Should be verifying that the requested collection is part of the database.  Using db.GetDatabaseCollection to do this check will also let you use GetCollectionID() with the response to get the collectionID, instead of fetching the manifest again.